test/langtools/tools/javac/patterns/Exhaustiveness.java

```diff
@@ -640,4 +597,131 @@ private int test(S obj) {
                """);
     }
 
+    @Test
+    public void testExhaustiveTransitive(Path base) throws Exception {
+        doTest(base,
+               new String[]{"""
+                            package lib;
+                            public sealed interface S permits A, B {}
+                            """,
+                            """
+                            package lib;
+                            public final class A implements S {}
+                            """,
+                            """
+                            package lib;
+                            public abstract sealed class B implements S permits C, D {}
+                            """,
+                            """
+                            package lib;
+                            public final class C extends B {}
+                            """,
+                            """
+                            package lib;
+                            public final class D extends B {}
+                            """},
+               """
+               package test;
+               import lib.*;
+               public class Test {
+                   private int test(S obj) {
+                       return switch (obj) {
+                           case A a -> 0;
+                           case C c -> 0;
+                           case D d -> 0;
+                       };
+                   }
+               }
+               """);
+    }
+
+    @Test
+    public void testExhaustiveIntersection(Path base) throws Exception {
+        doTest(base,
+               new String[]{"""
+                            package lib;
```

Comment:
it will be nice to have additional tests with conditional patterns (total, with a condition that is true, and not) to stress all the possible code paths

Reply:
Thanks Vicente! I've updated the patch based on your comments in https://github.com/openjdk/jdk17/pull/78/commits/a7a61692dcf3fbc8d24a5cf7c881ec635d654f62 .

What do you think?

Thanks!